### PR TITLE
e2e-runner: Update the image family used for GCI canary test images

### PR DESCRIPTION
**Pull request by wonderfly**
@Amey-D @spxtr Can you review?
@Random-Liu Heads up
cc/ @kubernetes/goog-image

### CODE REVIEW

**Comment by Random-Liu** [comment]
@wonderfly So I should also update my PR? :)

**Comment by wonderfly** [comment]
@Random-Liu Yes, and any other places where `gci-preview-test` is used.

File: hack/jenkins/e2e-runner.sh [hunk 1]
```diff
@@ -216,8 +216,8 @@ if [[ -n "${JENKINS_GCI_IMAGE_FAMILY:-}" ]]; then
   export KUBE_GCE_MASTER_PROJECT="${GCI_STAGING_PROJECT}"
   export KUBE_GCE_MASTER_IMAGE="$(get_latest_gci_image "${GCI_STAGING_PROJECT}" "${JENKINS_GCI_IMAGE_FAMILY}")"
   export KUBE_OS_DISTRIBUTION="gci"
-  if [[ "${JENKINS_GCI_IMAGE_FAMILY}" == "gci-preview-test" ]]; then
-    # The family "gci-preview-test" is reserved for a special type of GCI images
+  if [[ "${JENKINS_GCI_IMAGE_FAMILY}" == "gci-canary-test" ]]; then
```
Comment (Amey-D) [inline hack/jenkins/e2e-runner.sh]: curious - why do we need "-test" image here?  can we use "gci-canary" instead?

Comment (Amey-D) [inline hack/jenkins/e2e-runner.sh]: Ah sorry, I misread it the first time.  Makes sense now.

**Comment by Amey-D** [comment]
LGTM

**Pull request merged**